### DISCUSSION TITLE
Access to application definition

### DISCUSSION THREAD

**Discussion by ballesta** [TiagoSilvaPereira/vemto-issues#177] [Q&A] [answered]
Hi,

Question 1: Is it possible to access to application definition like:

- Get list of entities
- Get list of fields for each entity
- Get list of relations  for each entity
- Get list of App
- Get list of fields and presentation in app form
- ....

Question 2 : is it possible to generate a Vemto application from a higher level conceptual model?

Use case:
- Starting from an existing UML model : 
    - class diagram with classes, associations, inheritance, ...
    - Uml Use cases
    - ...
- Generate the corresponding Vemto application.

The goal would be to replace the interactive Vemto GUI usage and replace it by an automated tool.

Best regards
Bernard

**Comment by TiagoSilvaPereira** [comment, marked as answer]
Hi @ballesta,

Let me try to answer your questions:

**Question 1: Is it possible to access application definitions like...**

Not yet, it will be possible once I launch the **Plugins** integration, making it to get this data and use it in the way you want.

**Question 2: is it possible to generate a Vemto application from a higher level conceptual model?**

It will be probably possible after implementing the Plugins feature (you could create a plugin to import the UML data and create a Vemto schema), but not with the current Vemto version.

Cheers
